validator/sawtooth_validator/journal/consensus/dev_mode/dev_mode_consensus.py

```diff
@@ -35,16 +37,33 @@ class BlockPublisher(BlockPublisherInterface):
      DevMode Consensus (BlockPublisher) will read these settings
      from the StateView when Constructed.
     """
-
-    def __init__(self, block_cache, state_view, batch_publisher, data_dir):
-        super().__init__(block_cache, state_view, batch_publisher, data_dir)
+    def __init__(self,
+                 block_cache,
+                 state_view_factory,
+                 batch_publisher,
+                 data_dir):
+        super().__init__(
+            block_cache,
+            state_view_factory,
+            batch_publisher,
+            data_dir)
 
         self._block_cache = block_cache
-        self._state_view = state_view
+        self._state_view_factory = state_view_factory
 
         self._start_time = 0
         self._wait_time = 0
-        config_view = ConfigView(self._state_view)
+
+        # Using the current chain head, we need to create a state view and
+        # then from that a config view so we can get consensus configuration
+        # values
+        state_root_hash = \
+            self._block_cache.block_store.chain_head.state_root_hash if \
+            self._block_cache.block_store.chain_head is not None else \
+            INIT_ROOT_KEY
```

Comment:
Same goes here.  I need to go back and fix this to account for genesis block hack until we have genesis consensus.